src/Panel/Panel.php

```diff
@@ -89,14 +543,160 @@ public static function link(App $kirby): bool
         return true;
     }
 
+    /**
+     * Returns the page object for the given id
+     *
+     * @param string $id Page's id
+     * @return \Kirby\Cms\Page|null
+     * @throws \Kirby\Exception\NotFoundException if the page cannot be found
+     */
+    public static function page(string $id)
+    {
+        $id   = str_replace('+', '/', $id);
+        $page = kirby()->page($id);
+
+        if ($page && $page->isReadable() === true) {
+            return $page;
+        }
+
+        throw new NotFoundException([
+            'key'  => 'page.notFound',
+            'data' => [
+                'slug' => $id
+            ]
+        ]);
+    }
+
+    /**
+     * Returns the model's object for the given path
+     *
+     * @param string $path Path to parent model
+     * @return \Kirby\Cms\Model|null
+     * @throws \Kirby\Exception\InvalidArgumentException if the model type is invalid
+     * @throws \Kirby\Exception\NotFoundException if the model cannot be found
+     */
+    public static function parent(string $path)
+    {
+        $modelType  = in_array($path, ['site', 'account']) ? $path : trim(dirname($path), '/');
+        $modelTypes = [
+            'site'    => 'site',
+            'users'   => 'user',
+            'pages'   => 'page',
+            'account' => 'account'
+        ];
+
+        $modelName = $modelTypes[$modelType] ?? null;
+
+        if (Str::endsWith($modelType, '/files') === true) {
+            $modelName = 'file';
+        }
+
+        $kirby = kirby();
+
+        switch ($modelName) {
+            case 'site':
+                $model = $kirby->site();
+                break;
+            case 'account':
+                $model = $kirby->user(null, $kirby->option('api.allowImpersonation', false));
+                break;
+            case 'page':
+                $id    = str_replace(['+', ' '], '/', basename($path));
+                $model = $kirby->page($id);
```

Comment:
This kind of duplicates `Panel::page()`, but with a slightly different behavior (it also converts spaces to slashes for some reason).

Reply:
You are totally right. I replaced it with Panel::page. I have no idea why we want to support spaces here but not in Panel::page. This is probably just some weird artefact.